src/main/java/Duke.java

```diff
@@ -1,10 +1,440 @@
+     * @param words The array of words that compose the input command
+     */
+    private static void handleTaskDone(Task[] tasks, String[] words) {
+        try {
+            if (words.length == 1) {
+                showMessage("Sorry, the input task number is missing, please try again! :(");
+            }
+            for (int i = 1; i < words.length; i++) {
+                //check if the input character after the word "done" is integer value
+                int taskNumber = Integer.parseInt(words[i]);
+                if (taskNumber <= Task.totalTask && taskNumber > 0) {
+                    showTaskDoneMessage(tasks, taskNumber);
+                } else {
+                    showMessage("Sorry, the input task number is invalid, please try again! :(");
+                }
+            }
+        } catch (NumberFormatException ex) {
+            showMessage("Sorry, the input task number is invalid, please try again! :(");
+        }
+    }
+
+    /**
+     * Shows the formatted message string
+     *
+     * @param message The message to print
+     */
+    private static void showMessage(String message) {
+        System.out.print("\t@");
+        for (int i = 0; i < message.length() + 4; i++) {
+            System.out.print("-");
+        }
+        System.out.println("@");
+        System.out.println("\t   " + message);
+        System.out.print("\t@");
+        for (int i = 0; i < message.length() + 4; i++) {
+            System.out.print("-");
+        }
+        System.out.println("@");
+    }
+
+    /**
+     * Shows the message to indicate that the task is marked as done
+     *
+     * @param tasks      The array which stores all the tasks
+     * @param taskNumber The given task number to mark as done
+     */
+    private static void showTaskDoneMessage(Task[] tasks, int taskNumber) {
+        if (!tasks[taskNumber - 1].getDone()) {
+            tasks[taskNumber - 1].setDone();
+            System.out.println("\tHooray! Task number " + taskNumber + " has been marked completed!");
+            System.out.println("\t[✔] " + tasks[taskNumber - 1].getTask());
+        } else {
+            System.out.println("\tThe task number " + taskNumber + " - \"" + tasks[taskNumber - 1].getTask() + "\" has already been done!");
+        }
+    }
+
+    /**
+     * Prints the to-do list with frames
+     *
+     * @param tasks     the array of class Task instance which stores all the tasks added by the user
+     * @param stopIndex the last index of the array that is not null
+     */
+    private static void printToDoList(Task[] tasks, int stopIndex, int maxLength) {
+        //format to-do list
+        System.out.print("\t/");
+        if (maxLength < " My to-do list: ".length()) {
+            maxLength = " My to-do list".length();
+        }
+        for (int i = 0; i < maxLength + "| [ ][ ] 100. ".length(); i++) {
+            System.out.print("-");
+        }
+        System.out.println("\\");
```

Comment:
Could you consider adding comments to make what this function is doing clearer?

Reply:
Sure! I will consider improving the code readability for this printToDoList method, so it basically just print a to-do list with frames like:
/---------------------\
| [ ][ ] 1. Your Tasks... |
\---------------------/
Thanks for your suggestion!